include/boost/geometry/formulas/area_formulas.hpp

```diff
@@ -407,93 +409,107 @@ class area_formulas
 
-        CT alp1 = i_res.azimuth;
-        CT alp2 = i_res.reverse_azimuth;
+        CT const alp1 = i_res.azimuth;
+        CT const alp2 = i_res.reverse_azimuth;
 
         // Constants
 
         CT const ep = spheroid_const.m_ep;
-        CT const f = formula::flattening<CT>(spheroid_const.m_spheroid);
-        CT const one_minus_f = CT(1) - f;
-        std::size_t const series_order_plus_one = SeriesOrder + 1;
-        std::size_t const series_order_plus_two = SeriesOrder + 2;
+        CT const one_minus_f = CT(1) - spheroid_const.m_f;
 
         // Basic trigonometric computations
-
-        CT tan_bet1 = tan(get_as_radian<1>(p1)) * one_minus_f;
-        CT tan_bet2 = tan(get_as_radian<1>(p2)) * one_minus_f;
-        CT cos_bet1 = cos(atan(tan_bet1));
-        CT cos_bet2 = cos(atan(tan_bet2));
-        CT sin_bet1 = tan_bet1 * cos_bet1;
-        CT sin_bet2 = tan_bet2 * cos_bet2;
-        CT sin_alp1 = sin(alp1);
-        CT cos_alp1 = cos(alp1);
-        CT cos_alp2 = cos(alp2);
-        CT sin_alp0 = sin_alp1 * cos_bet1;
+        // the compiler could optimize here using sincos function
+        // TODO: optimization: those quantities are already computed in inverse formula
+        // at least in some inverse formulas, so do not compute them again here
+        /*
+        CT sin_bet1 = sin(lat1r);
+        CT cos_bet1 = cos(lat1r);
+        CT sin_bet2 = sin(lat2r);
+        CT cos_bet2 = cos(lat2r);
+
+        sin_bet1 *= one_minus_f;
+        sin_bet2 *= one_minus_f;
+        normalize(sin_bet1, cos_bet1);
+        normalize(sin_bet2, cos_bet2);
+        */
+
+        CT const tan_bet1 = tan(lat1r) * one_minus_f;
+        CT const tan_bet2 = tan(lat2r) * one_minus_f;
+        CT const cos_bet1 = cos(atan(tan_bet1));
+        CT const cos_bet2 = cos(atan(tan_bet2));
+        CT const sin_bet1 = tan_bet1 * cos_bet1;
+        CT const sin_bet2 = tan_bet2 * cos_bet2;
+
+        CT const sin_alp1 = sin(alp1);
+        CT const cos_alp1 = cos(alp1);
+        CT const cos_alp2 = cos(alp2);
+        CT const sin_alp0 = sin_alp1 * cos_bet1;
 
         // Spherical term computation
 
-        CT sin_omg1 = sin_alp0 * sin_bet1;
-        CT cos_omg1 = cos_alp1 * cos_bet1;
-        CT sin_omg2 = sin_alp0 * sin_bet2;
-        CT cos_omg2 = cos_alp2 * cos_bet2;
-        CT cos_omg12 =  cos_omg1 * cos_omg2 + sin_omg1 * sin_omg2;
         CT excess;
 
-        bool meridian = get<0>(p2) - get<0>(p1) == CT(0)
-              || get<1>(p1) == CT(90) || get<1>(p1) == -CT(90)
-              || get<1>(p2) == CT(90) || get<1>(p2) == -CT(90);
+        auto const half_pi = math::pi<CT>() / 2;
+        bool meridian = lon2r - lon1r == CT(0)
+            || lat1r == half_pi || lat1r == -half_pi
+            || lat2r == half_pi || lat2r == -half_pi;
 
-        if (!meridian && cos_omg12 > -CT(0.7)
-                      && sin_bet2 - sin_bet1 < CT(1.75)) // short segment
+        if (!meridian && (i_res.distance) < 10000)  // short segment
```

Comment:
Two problems with it:
- it's looks like an arbitrary number based on an (informed) guess but instead should probably be empirically found by experimenting
- it should at least depend on the size of the globe, it is possible to create a spheroid using km instead of meters for axes, in such case this number should probably be `10`, not to mention that the flattening is probably also a factor

Reply:
> it's looks like an arbitrary number based on an (informed) guess but instead should probably be empirically found by experimenting

see https://github.com/boostorg/geometry/pull/801#discussion_r570871224 above, the experiments are with WGS84, of course one could experiment more, but for the purpose of this PR I think is enough

> it should at least depend on the size of the globe, it is possible to create a spheroid using km instead of meters for axes, in such case this number should probably be 10, not to mention that the flattening is probably also a factor

right, I replaced `10000` with `mean_radius<CT>(spheroid_const.m_spheroid) / CT(638)`